moveit_kinematics/kdl_kinematics_plugin/src/chainiksolver_vel_mimic_svd.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright  (C)  2007  Ruben Smits <ruben dot smits at mech dot kuleuven dot be>
+
+// Version: 1.0
+// Author: Ruben Smits <ruben dot smits at mech dot kuleuven dot be>
+// Maintainer: Ruben Smits <ruben dot smits at mech dot kuleuven dot be>
+// URL: http://www.orocos.org/kdl
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+// Modified to account for "mimic" joints, i.e. joints whose motion has a
+// linear relationship to that of another joint.
+// Copyright  (C)  2013  Sachin Chitta, Willow Garage
+
+#include <moveit/kdl_kinematics_plugin/chainiksolver_vel_mimic_svd.hpp>
+#include <ros/console.h>
+
+namespace
+{
+unsigned int countMimicJoints(const std::vector<kdl_kinematics_plugin::JointMimic>& mimic_joints)
+{
+  unsigned int num_mimic = 0;
+  unsigned int index = 0;
+  for (const auto& item : mimic_joints)
+  {
+    if (item.map_index != index)
+      ++num_mimic;
+    ++index;
+  }
+  return num_mimic;
+}
+}
+
+namespace KDL
+{
+ChainIkSolverVelMimicSVD::ChainIkSolverVelMimicSVD(const Chain& chain,
+                                                   const std::vector<kdl_kinematics_plugin::JointMimic>& mimic_joints,
+                                                   bool position_ik, double threshold)
+  : mimic_joints_(mimic_joints)
+  , num_mimic_joints_(countMimicJoints(mimic_joints))
+  , chain_(chain)
+  , jnt2jac_(chain)
+  , svd_(position_ik ? 3 : 6, chain_.getNrOfJoints() - num_mimic_joints_, Eigen::ComputeThinU | Eigen::ComputeThinV)
```

Comment:
please add comment describing this logic... its very difficult to understand having it nestled in the initalizer list. alternatively, providing a anon helper function to call within the initializer list would be better

Reply:
Done.